src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.CompilationTracker.cs

```diff
@@ -182,8 +183,16 @@ public CompilationTracker FreezePartialStateWithTree(SolutionState solution, Doc
                 // The user is asking for an in progress snap.  We don't want to create it and then
                 // have the compilation immediately disappear.  So we force it to stay around with a ConstantValueSource.
                 // As a policy, all partial-state projects are said to have incomplete references, since the state has no guarantees.
-                return new CompilationTracker(inProgressProject,
-                    new FinalState(new ConstantValueSource<Optional<Compilation>>(inProgressCompilation), inProgressCompilation, hasSuccessfullyLoaded: false));
+                // We also will drop our generator driver, since the in-progress state will simply have to be whatever we already had and
+                // we won't run any generators any further.
```

Comment:
so, does tihs mean in-progress will never see any generator files?  would that make in-progress almost always useless?  what about preserving the generated files from the previous state and presuming they're the generated code for the in-progress state?

Reply:
If something has already pulled the compilation forward to the FinalState then the final state should be used and we're good but yeah that's not great. I'm going to need to redo the GeneratorDriver API to be more explicit about the syntax trees which will then let us carry those along accordingly.

Reply:
I will add a test though that fetching the full compilation and then fetching an in-progress does still have generators, and also will manually test we're not seeing really bad behavior in completion...

Reply:
Test was added for that.